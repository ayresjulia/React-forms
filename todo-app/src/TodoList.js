import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
	const [ todos, setTodos ] = useState([]);
	const addTodo = (newTodo) => {
		setTodos((todos) => [ ...todos, { ...newTodo, id: uuid() } ]);
	};
	const removeTodo = (id) => {
		setTodos((todos) => todos.filter((todo) => todo.id !== id));
	};

	return (
		<div className="App">
			<h3>Todos</h3>
			<div className="TodoList-form">
				<NewTodoForm addTodo={addTodo} />
			</div>
			<div className="TodoList">
				<ul>
					{todos.map(({ id, todo }) => (
						<Todo key={id} id={id} removeTodo={removeTodo} todo={todo} />
					))}
				</ul>
			</div>
		</div>
	);
};

export default TodoList;
