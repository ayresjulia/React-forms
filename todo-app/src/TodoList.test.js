import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", () => {
	render(<TodoList />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

it("should add new todo item", () => {
	const { queryByText, getByLabelText } = render(<TodoList />);
	const input = getByLabelText("New Todo");
	const btn = queryByText("Add");
	expect(queryByText("walk the dog")).not.toBeInTheDocument();
	fireEvent.change(input, { target: { value: "walk the dog" } });
	fireEvent.click(btn);
	expect(queryByText("walk the dog")).toBeInTheDocument();
});

it("should delete new todo item", () => {
	const { queryByText, getByLabelText } = render(<TodoList />);
	const input = getByLabelText("New Todo");
	const btn = queryByText("Add");
	expect(queryByText("walk the dog")).not.toBeInTheDocument();
	fireEvent.change(input, { target: { value: "walk the dog" } });
	fireEvent.click(btn);
	expect(queryByText("walk the dog")).toBeInTheDocument();
	fireEvent.click(queryByText("X"));
	expect(queryByText("walk the dog")).not.toBeInTheDocument();
});
