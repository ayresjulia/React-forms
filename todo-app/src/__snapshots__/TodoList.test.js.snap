// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h3>
      Todos
    </h3>
    <div
      class="TodoList-form"
    >
      <form
        class=""
      >
        <div
          class="form-row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="form-group"
            >
              <label
                class=""
                for="todo"
              >
                New Todo
              </label>
              <input
                class="form-control"
                id="todo"
                name="todo"
                type="text"
                value=""
              />
            </div>
            <button
              class="btn btn-secondary"
            >
              Add
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="TodoList"
    >
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
